engine/apps/mobile_app/tasks/new_alert_group.py

```diff
@@ -0,0 +1,139 @@
+    apns_volume = mobile_app_user_settings.important_notification_volume if critical else None
+    message_type = MessageType.IMPORTANT if critical else MessageType.DEFAULT
+    apns_sound_name = mobile_app_user_settings.get_notification_sound_name(message_type, Platform.IOS)
+
+    fcm_message_data: FCMMessageData = {
+        "title": alert_title,
+        "subtitle": alert_subtitle,
+        "orgId": alert_group.channel.organization.public_primary_key,
+        "orgName": alert_group.channel.organization.stack_slug,
+        "alertGroupId": alert_group.public_primary_key,
+        # alert_group.status is an int so it must be casted...
+        "status": str(alert_group.status),
+        # Pass user settings, so the Android app can use them to play the correct sound and volume
+        "default_notification_sound_name": mobile_app_user_settings.get_notification_sound_name(
+            MessageType.DEFAULT, Platform.ANDROID
+        ),
+        "default_notification_volume_type": mobile_app_user_settings.default_notification_volume_type,
+        "default_notification_volume": str(mobile_app_user_settings.default_notification_volume),
+        "default_notification_volume_override": json.dumps(
+            mobile_app_user_settings.default_notification_volume_override
+        ),
+        "important_notification_sound_name": mobile_app_user_settings.get_notification_sound_name(
+            MessageType.IMPORTANT, Platform.ANDROID
+        ),
+        "important_notification_volume_type": mobile_app_user_settings.important_notification_volume_type,
+        "important_notification_volume": str(mobile_app_user_settings.important_notification_volume),
+        "important_notification_volume_override": json.dumps(
+            mobile_app_user_settings.important_notification_volume_override
+        ),
+        "important_notification_override_dnd": json.dumps(mobile_app_user_settings.important_notification_override_dnd),
+    }
+
+    number_of_alerts = alert_group.alerts.count()
+    apns_payload = APNSPayload(
+        aps=Aps(
+            thread_id=thread_id,
+            badge=number_of_alerts,
+            alert=ApsAlert(title=alert_title, subtitle=alert_subtitle),
+            sound=CriticalSound(
+                # The notification shouldn't be critical if the user has disabled "override DND" setting
+                critical=overrideDND,
+                name=apns_sound_name,
+                volume=apns_volume,
+            ),
+            custom_data={
+                "interruption-level": "critical" if overrideDND else "time-sensitive",
+            },
+        ),
+    )
+
+    return construct_fcm_message(message_type, device_to_notify, thread_id, fcm_message_data, apns_payload)
+
+
+@shared_dedicated_queue_retry_task(autoretry_for=(Exception,), retry_backoff=True, max_retries=MAX_RETRIES)
+def notify_user_async(user_pk, alert_group_pk, notification_policy_pk, critical):
```

Comment:
it would make sense to rename this task to something more descriptive (ex. `notify_user_about_alert_group`). 

What is the order of operations for renaming a task? Can I simply rename it, leave the [old task routing definition](https://github.com/grafana/oncall/blob/dev/engine/settings/prod_without_db.py#L115) in place + add the new one, then remove the old one in a future release?

Reply:
I would create the new one, add it to routing and call wherever it's needed and leave the old, just change its code to call the new task. Then the old one can be removed in a future release.